### ISSUE TITLE
Wiisxr issue list. 

### ISSUE THREAD

**Issue by Orvis25** [Mystro256/wiisxr#13]
I do not know if these help make the program better, but here is a list of issued i am having with the current version you have available. 

Out of a about 15 games i currently play, two are giving me hell. 
Vigilante 8 and Vigilante 8 second offence.

Vigilante 8 loads up, and has pretty fine visual, but the music plays all messed up, and in adition there is some lagging going on

Vigalante 8 Second offense crashes the system to a dump file.

Other issues i am having: 

I have my roms and save folder on a USB HDD. When i set my bios to load form usb, and all my other setting, I hit save settings. But when i try to exit Wiisxr my wii just closes the app, and remains on a black screen (power button stays green) never going back to wii main menu. So i have to hard reset the wii AND my external HHD(hold wii power down till it turns red "off", then unpllug hdd power so it spins down, then plug back in. So the next time I open Wiisxr the saved setting do not automatically load and i have to put them all back in again (which is really annoying for the controller keys)

I also have an issue with when i hit the reset button to back out of the game to wiisxr menu, i get an error that my USB drive is no longer available, and the only way i can reset it is shutting down my wii, and restarting. It someone like "ejects" my hdd form the wii somehow for no reason when backing out via reset button. 

I think most of these USB problems is with whatever ui code or whatever is used for wii64, wiisx, FCE Ultra GX, Snes9x GX, and VBA GX. Reason being is because i have similar usb "disconnecting" errors in ALL of those emulators, who all use similar looking UI configurations (so i assume based off same thing). Now i know it is not my drive because if i use something  like USB Loader GX, i can play for HOURS with no issues, and can switch game at will. SO i know it is something with the coding between all those emulators, while USB loader is fine.

Anyway, I hope this helped. 

ps: picture is the dump crash image i get for vigilante 8 second offence

![dsc_0001 1](https://cloud.githubusercontent.com/assets/16249891/13694165/2c1dd982-e718-11e5-9dbb-d696efdb2488.JPG)

**Comment by Mystro256** [comment]
Indeed, a lot of these issues I'm aware of but having them here may help if someone else wants to try to fix it.

I'm mostly working on fixing up the code: e.g. cleaning up, formatting, style issues, some memory leaks, used variables, performance issues. Compatibility is a little bit harder for me to work on. I plan to redo or work on the sound after I pull my **experimental-fixes** back into **master**.

As well, I'll look into the USB issues... this one has been bothering me for a while, but it helps that you can narrow it down to a likely a library issue (i.e. something USB Loader GX doesn't use). It could be related to a memory leak that I'm trying to fix.

**Comment by roflcopter777** [comment]
If you need any help testing builds and providing feedback, I'd be willing to do so

**Comment by Orvis25** [comment]
Well glad i was able to narrow it down for you mysto256.

**Comment by Mystro256** [comment]
@roflcopter777 
I make experimental changes to the experimental-fixes branch. If you want to do a regression test every time I make a build, please feel free to do that. You can look at all the builds by going here:

https://github.com/Mystro256/wiisxr/commits/experimental-fixes/Gamecube/WiiSX.dol

If you find a regression, please make note of it by leaving a comment on commit that it was introduced. 

In case you're unfamiliar with Github, go to the link above, and click on one of the commits (such as "Minor coding error fixes with rebuild" for example). At the bottom, you can leave a comment, or download the build by going to the entry that's called "Gamecube/WiiSX.dol", clicking the View button, then clicking Raw button.

@Orvis25
I'm always up for some help. I would advise your friend to just look at the code here. I personally would prefer help by means of issues, comments, patches or pull requests (if your friend knows how to use git/Github).

**Comment by roflcopter777** [comment]
Awesome, I'll be sure to test those builds out, anything I should test in particular? I also noted that Grandia doesn't seem to boot, but locks up the emulator, but I'll retest it later :) Thank you for your work  ^^

**Comment by Orvis25** [comment]
here is another crash bump picture for vigilante 8 when loading levels.

Do these actually provide you with useful info? cuz i can keep posting them if they help.
![dsc_0002 1](https://cloud.githubusercontent.com/assets/16249891/13726090/25b23104-e87d-11e5-8311-fb016c6c1afd.JPG)

**Comment by Mystro256** [comment]
@roflcopter777 Regressions are most important. I don't want to merge the changes until I know it doesn't break anything. All the changes I've made have been with a static code analyzer and walking through the code by hand. I have yet to have time to legitimately test it in the last 6 months.

@Orvis25 I'm not sure yet. I mean it's no harm for now, but I'm more interested in how to reproduce these errors, and any insight if applicable.
For example, useful insight would be a change in behaviour with different settings or perhaps if it's not reproducible with the PCSX, PCSXR, or another open source PlayStation emulator, if available. If you test it on PCSX or another, make sure you don't use an opengl or sdl plugin to test it, you use 32bit _not_ 64bit, and you specify your setup.

**Comment by roflcopter777** [comment]
I don't know if the ISO dump I have is bad or what, it doesn't crash the emulator, i merely shows a blank screen then kicks me out to the Homebrew Menu, no other games do that, will test again.

**Comment by Orvis25** [comment]
@Mystro256 If this helps i found out that ePSXe can emulate vigilante 8 successfully. 
https://www.youtube.com/watch?v=txzyxerrWZ8

Maybe there is some code in there you can see that will fix the problem with WiiSXr ??? 

Hope that helped :)

Edit: It seems ePSXe is still receiving updated for better compatibility even in 2016. Would it be easier to modify ePSXe to work on a wii then trying to repair and update wiiSXr?

**Comment by Mystro256** [comment]
@roflcopter777 I would recommend using Wiiload and testing each build. I could have easily broke something down the line, i.e. a regression.

@Orvis25 unfortunately ePSXe is closed source :(

**Comment by Orvis25** [comment]
@Mystro256 I do not suppose asking them would help? 

Maybe if they see you are developing/coding for a wii and need a better emulator they will let you use the code?

Always worth a shot no?

**Comment by roflcopter777** [comment]
Wiiload? I don't know what that is. Wiiloader?

**Comment by Orvis25** [comment]
@roflcopter777 Google man.... Use it..... Literally first link....
http://onyxsoft.se/wiiloader.html

**Comment by Mystro256** [comment]
Read up on it here:
http://wiibrew.org/wiki/Wiiload

There's Mac, windows, Linux versions

**Comment by roflcopter777** [comment]
@Orvis25  Gee, so sorry for asking -_-  I don't have that app, so..yeah...anyways. The issue has been plaguing the emulator for some time though, even before the recent revision.

**Comment by Mystro256** [comment]
Wiiload is just an app on your computer where you can send the dol file to
the homebrew channel over wifi without having to put it on an sdcard or USB
stick. I use it to test lots of builds at once.

I'm working on the emulator whenever I can and once I an confirm there's no
regressions in the experimental builds, I can push to changes to the main
branch. The problem is that I have no time to test this stuff.

**Comment by roflcopter777** [comment]
Ah, well, I never tried that app before, I'll give it a whirl, my ISO is probably corrupted, I'll try again, my apologies -_-

**Comment by Orvis25** [comment]
@Mystro256 Have you ever asked if the ePSXe team would let you developer their emulator for wii?

**Comment by Mystro256** [comment]
@Orvis25 honestly I would rather not work on proprietary code if I had the choice. It's a bit of an ethical issue; if epsxe has no interest in releasing their source code, I have no interest in contributing to them.

I recently found out about libretro; I would like to look into it before continuing development.

**Comment by Orvis25** [comment]
@Mystro256 
so does that mean that realistically, we wont see better game comparability, just bug fixes with its current comparability?

**Comment by roflcopter777** [comment]
Where did you get that assumption from?

**Comment by Orvis25** [comment]
because wiisx has not been updated since like 2013, and wiisxr is based on it. And mystro256 said he is working on things like memory leaks and some bug fixes with usb, not actually improving game compatibility..

**Comment by roflcopter777** [comment]
Well, gee, maybe if other people were willing to help him out, that would be different.

**Comment by Mystro256** [comment]
Honestly I have interest in improving the emulator in general, but I'm very
limited in my time to program it. I would hope that I can either improve
compatibility eventually or my revival of the emulator can encourage people
to help. Hence why its on github, as it makes it very accessible to amateur
programmers (forking, pull requests, patches, issue tracking)

**Comment by Orvis25** [comment]
@roflcopter777 Would be nice. i cant program worth a crap though :P Took an HTML class in HS and got like a 71%. programming and me do not work. but i always willing to bug test stuff. 

@Mystro256 Oh, well cool. make sure you work on vigalante 8 games :P really would love to play that again, and it does not work on ps1 emulator, or n64 emulator  for wii :(

**Comment by roflcopter777** [comment]
What can I do to help attract other programmers?

**Comment by Orvis25** [comment]
@Mystro256 
Question. If you fix the memory bug in wiiSX, will it carry over to things like wii64, FCE Ultra GX, Snes9x GX, and VBA GX? you said previously they all use the same loader or whatever. that why they ALL have the same issue with usb. 

Just curious if you fix it here, if you will take over and update those (cuz they are abandoned atm. Last update was like 2012)

**Comment by Mystro256** [comment]
@roflcopter777 Honestly I'm not sure.

@Orvis25 Honestly it depends on where the issue is. It may be a simple issue that can be fixed in upstream libraries and thus just will just require a rebuild in these projects... but if it's something more complicated requiring a work around a quirk in the Wii system, mileage may vary. I will try to port a fix if possible, but I still need to figure it out first.

**Comment by Mystro256** [comment]
Anyone able to see if I broke anything with the following build? Most games from I can see are either the same or slightly faster with the latest changes:

https://github.com/Mystro256/wiisxr/raw/experimental-fixes/Gamecube/WiiSX.dol

**Comment by roflcopter777** [comment]
Gonna test it out today and let you know :)

**Comment by Orvis25** [comment]
@Mystro256 If i get some time, i will test it out over a period of a few hours.

**Comment by Orvis25** [comment]
@Mystro256 all i do is replace the old boot.dol with this file right? (i just del old of sd, drag this, rename to boot?)

is that correct?

**Comment by roflcopter777** [comment]
Yes, you just replace the boot.dol file is all

**Comment by Orvis25** [comment]
@Mystro256 
Just tried the following games using the same settings i did before (mainly set bios to usb, boot through bios to yes, rebind all keys for gamecube controller, and that's it) 

 These will be the only games i will be testing, as they are the only ones i want to play:

Vigilante 8 -> freezes at menu and i get a "EEEEEEEEEEEEEEEEEEEE" sound. have to hard reset
Vigilante 8 Second Offence ->  Same as above
Twisted medal -> Same as above
Twisted medal 2 -> Same
Twisted medal 3 -> Same
Twisted medal 4 -> Same

It also still will not "exit" back to the wii menu (granted i do not try to load a game that is). Still just black screens and hangs until i hard reset. So i can never "save" any settings because every time i hard reset, and reopen wiisxr, it never "loads" the save file, so i have to manually reset my key bindings for the gamecube controller EVERY TIME. 

Conclusion: This update completely broke the emulator

I restored the original boot.dol and its working again (with all same issues as previous version obviously)

ps: please, edit the "credits" button so it actually says wiisxr v. 1.xxxx or something. I do not like that it says its still wiisx beta 2.1. Would also make troubleshooting easier in future in case someone is not up to date or something :D

**Comment by Mystro256** [comment]
@Orvis25

I'll try to reproduce this and fix it before I merge in back into master.

@roflcopter777 

Let me know if you find any other issues.

**Comment by roflcopter777** [comment]
@Mystro256  Just tried Crash Bandicoot 3.bin, emulator crashed with a very high-pitched buzzing sound. Crash Bandicoot 2 also crashes, they worked on the previous build BTW, so it's not my ISO. Should I have deleted or changed something in the folder or made a new folder? Why's it crashing? Using a Wii U also, and these ISO images loaded fine beforehand -_-

Don't know what I'm doing wrong, previous build ran these fine, now I can't load any ISO images without that horribly high-pitched buzzing. 

Edit: Deleted old config file, made new wiiSX folder. If this doesn't work, I don't freaking know what to do :(

**Comment by roflcopter777** [comment]
@Mystro256  Nope, not a damn thing. Changed BIOS from HLE to SD, nothing, games still crash, buzzing sound, Breath of Fire 3 showed a black screen, buzzing, Crash Bandicoot games lock up the menu. I don't know what I'm doing wrong, I made a new folder, I made a new config, I don't know what to do to, I thought I was being helpful, but now, I've made it worse, because I'm the only one this is happening to, no one else will be able to reproduce this error, and others will say "Oh it must be your end." So, yeah, I can't get ANY game to work -_- \* sigh\* 

Maybe it's the fact I'm using a Wii U and not a Wii, but that shouldn't effect it or be relevant, I don't know. Dammit...

Edit: Seems @Orvis25  is having the same issue, how come we can't load any games at all?

**Comment by Mystro256** [comment]
@roflcopter777 
No, you didn't do anything wrong, I was just looking to see if I broke something, and it seems I did. This is what I mean by "regression testing"... Sometimes when you try to fix something, you end up breaking something else. Knowing which games broke will help me fix the issues, as I can try to reproduce the issues on my system as well.

**Comment by roflcopter777** [comment]
@Mystro256 I was looking forward to seeing the speed improvements in the games _sigh_ I hope I can soon. This has got to be stressful for ya. Still...hopefully it's not a big regression -_-

**Comment by Mystro256** [comment]
We'll see, worst case I undo what I did.

**Comment by roflcopter777** [comment]
And have the speed not improved? Ugh....

**Comment by Orvis25** [comment]
@Mystro256 Just keep at it man. We are rooting for you :D !!! 

do not give up and let this project die again!

**Comment by Mystro256** [comment]
@roflcopter777
Lol dude, it's going to take a while if I'm the only one working on it. Rome wasn't built in a day.

@Orvis25
Thanks! Glad to hear the enthusiasm :))

**Comment by Orvis25** [comment]
@Mystro256 Maybe try starting with something smaller. Like fixing the fact it does not "exit" to the wii menu. And modifying the credit to now say wiisxR and add yourself in there. 

Small stuff first :)

**Comment by Mystro256** [comment]
@Orvis25
Haha of course, the programming isn't an issue, I actually do this for a living (I'm a computer engineer at a firm), it's just lack of time.

As for the exit issue... Unfortunately I'm not sure what's causing the issue as I looked into what I thought was causing it... But it isn't the case. I thought fixed it in the following build but I'm not sure:
https://github.com/Mystro256/wiisxr/raw/master/Gamecube/WiiSX.dol

If you use that build and outline exactly what you do to cause the issue (including your setup and steps) I could _"possibly_\* pin point the issue. Honestly I'm a little stumped haha.

**Comment by Orvis25** [comment]
@Mystro256 Does it close correctly for you? 

All i do is install wiisxr on sd, launch wiisxr via channel icon or via homebrew. It does not matter because when i go to "exit" it just instantly black screens, locks up, and i have to hard reset.

idk what other info i can give.

**Comment by Mystro256** [comment]
@Orvis25 
OH that issue, I fixed that like a month ago. If you want to test it, use this build, NOT the one I broke:
https://github.com/Mystro256/wiisxr/raw/master/Gamecube/WiiSX.dol

My "risker" builds are kept here (the one that broke the emulator, the other one is fine though):
https://github.com/Mystro256/wiisxr/raw/experimental-fixes/Gamecube/WiiSX.dol

But once I finish a few things I plan to start releasing zip files for easier testing... my current organization is messy and confusing.

**Comment by Orvis25** [comment]
@Mystro256 

is the "fixed" one (first link) NEWER than the one i currently use then? cuz i downloaded the one from your thing when i found out about wiisxr (like last month or whatever)

**Comment by Orvis25** [comment]
@roflcopter777 
Dont be a jerk man. i would love to see you pick up a 4 year old dead project programmed by someone else, and program speed improvements, cleaning code, and improving stability fixes into something you never wrote, doing so in only your spare time while managing a full time job, and a family. 

I for one am glad to see at least an attempt being made. Be it breaking things before fixing, at least and attempt never the less.

**Comment by roflcopter777** [comment]
@Mystro256  Sorry about the other day -_-

**Comment by OronFortune** [comment]
While i am just a step or two above a complete noob if there is something i could do to help and i know or can be taught to do it i have time to help

**Comment by Mystro256** [comment]
@OronFortune 
I'd like to start a compatibility wiki to help me pinpoint which games don't work... not sure how wiki's work on Github though. I believe you fork me, make a change in the wiki tab for your fork then do a pull request to pull the changes into here.

Maybe we can use the wiisx compatibility chart as a starting point.

**Comment by Mystro256** [comment]
I made a page, see if you can edit it or not. I believe I turned off restricted editing (should still require a GitHub account though).
https://github.com/Mystro256/wiisxr/wiki/Compatibility

**Comment by roflcopter777** [comment]
@Mystro256  Awesome, gonna test this today ^^ So is there a particular format we use for the text,  like using tables and so on like a wiki page?

Edit: I see something maybe like this as a template?

==A==
{{Table Start}}
| name =  | region = USA/EUR/JPN | status = Doesn't Work/Works/Issues | notes = please add a note if needed}}

Maybe? I'm not too good at formatting text, but something I saw on a GBA Temp Wiki.

**Comment by Mystro256** [comment]
@roflcopter777 
I haven't exactly decided, but as long as it's coherent and easy to read. You can bring the data straight from gbatemp, but also specify the version it was tested on (E.g. wiisxr 2.2, wiisx mod 5, mod 4, ..., wiisx 2.1, etc.)

Maybe we could use the gbatemp wiki instead, like they did with not64?... I just don't like how I don't know what version it was tested with, plus I'm not sure if they want to keep wiisx separate from wiisxr.

**Comment by roflcopter777** [comment]
Anyone can make wiki entries, I made one for Virtual Console injections for the Wii U and they didn't seem to mind, hmm, I can definitely try something, but at the same time, I don't want to detract from GitHub either, so I'm conflicted. The only caveat is you do need to make a GBA Temp Wiki account, which is simple, I can take a look at it.

**Comment by Mystro256** [comment]
Honestly if we make a new page on gbetemp, or do what they did with not64,
it maybe more accessible to people. I just want the source on github, I
could care less about the rest. I should really make a gbatemp or GC
forever forum post, I'll try to remember to do it this weekend.

**Comment by Orvis25** [comment]
@Mystro256 Honestly, making a pub list is prob going to FLOOD you with games that "dont work" 

Maybe wait a few versions before doing like a post on a more pub area. Improve and fix a few more minor things, and fix the "credits" section so it reflects wiisxR with YOUR version number.

just my 2 cents

**Comment by roflcopter777** [comment]
@Mystro256 Going to at least start a template as I've figured out a decent template with table listing the game name, the region, compat, notes and so on. Gonna show a preview in a bit. 

Edit: Or  not, proving to be difficult to set up, ugh.

**Comment by Mystro256** [comment]
@Orvis25 
The credits section wasn't updated? It should have been updated yesterday. I just posted a new 2.2 version in the release section (this is the first actual build that I did this morning, the previous ones were all test builds):
https://github.com/Mystro256/wiisxr/releases

@roflcopter777 
Honestly as long as you find something that works, I could care less. I realize that Github's wiki's maybe a little limited. I made a template to work off of.

**Comment by Orvis25** [comment]
@Mystro256 Is this the one that messed up and broke everything? Or a newer one with that problem fixed?

**Comment by Mystro256** [comment]
@Orvis25 
It's the one that works but with some minor improvements, complete rebranding to WiiSXR (including folders) and I updated all the version numbers/credits. I plan to only release stable builds, rather than unstable/development builds, which is what I was doing before.

**Comment by roflcopter777** [comment]
@Mystro256  Awesome, I'll work off the GitHub Wiki, it looks more compact but way more organized, I have several games to test :)

**Comment by Orvis25** [comment]
@Mystro256 Ok, got it installed, changed folder locations on usb. 

still gave one problem that i would love fixed APSP:

If i go to settings -> general -> and set bios to usb and boot through bios to yes, then save setting to the usb, the next time i open program (after closing it successfully without crash), it does not auto load the settings so i need to set them manually again. 

Edit: PS. games are about the same as they were before. just to let you know.

**Comment by Mystro256** [comment]
@Orvis25 
Indeed, I'm aware of this issue, I just haven't had time to look into it; there was a bunch of higher priority crash fixes that needed to be fixed. There's also a bug with loading ISO's from the SD card that needs to be fixed, once I figure out what's causing it because it's preventing me from fixing other issues. It may be related to the issue I spoke with you about last night.

@roflcopter777 
Make sure you only test with the builds listed on the release page (only 2.2 right now), all other builds I posted are no longer relevant:
https://github.com/Mystro256/wiisxr/releases

**Comment by Orvis25** [comment]
@Mystro256 Ah ok! well good to know! Good luck :) 

i chat with ya tonight.

**Comment by roflcopter777** [comment]
@Mystro256 So loading ISO from SD cards is not recommended as they don't work?  I'm confused. I do have a USB HDD I can test with it, not sure what else I should do.

**Comment by Mystro256** [comment]
@roflcopter777 
Sorry, let me clarify, the SD card filebrowser code seems to be fragile for some reason (prone to regression when I change other things around it)... I purposely tried not to break it for 2.2, but I couldn't make certain fixes because I didn't want to break SD support.

This is the issue I attempted to describe in #12 . I just figured out last night that it's caused by the SD card code, so progress is being made, albeit slowly.

**Comment by roflcopter777** [comment]
@Mystro256  That's fine, man, I understand, some games do seem to be loading fine, Crash Bandicoot and so on, I'll test as much as I can, thank you for your hard work ^^ I'll test as much as possible.

**Comment by Orvis25** [comment]
@Mystro256 
Is there a way to edit the default settings for the gamecube controller binding so i do not have to spend like 5 min changing them all every time i open the wiisxr app? 

Edit: like changing the file on sd manually because it does not save or load the save file currently

**Comment by roflcopter777** [comment]
@Mystro256  Updated compat list, couldn't get Chrono Cross to load for some reason, maybe a bad dump? Final Fantasy IX resulted in immediate code dump, which I took a screenshot of, not sure if the info in the code dump will prove useful, I can post a link to it.

**Comment by Mystro256** [comment]
@roflcopter777 
No need, I'll ask you for it if I can't reproduce it.
As for Chrono cross, I'm pretty sure you have a bad dump because it works for me.

**Comment by Orvis25** [comment]
@Mystro256 i take it no then to my previous question?

**Comment by Mystro256** [comment]
@Orvis25 
Sorry I don't know off the top of my head. I believe it prioritizes the SD config file, but you could probably edit it with a text editor if its not saving.

**Comment by roflcopter777** [comment]
@Mystro256  Very true, sorry for the false positive, will test another dump of it. Keep up the good work, man. Any word on why Final Fantasy IX crashes when I boot it up? 

This is the DSI exception/code dump information that was generated when I loaded the ISO
http://i.imgur.com/Jkwc43n.jpg

**Comment by OronFortune** [comment]
Ok was able to edit it and since i had the new build you put up i did a ff tactics info it might have been useful info. I have a few more games i can test and add info on but wanted to make sure that i did it right and i am not sure how to pull it back in as you say lol sorry

**Comment by roflcopter777** [comment]
No worries, yeah, it's pretty straightforward

**Comment by Orvis25** [comment]
@Mystro256 
After testing on the new version. i am currently still having problems with these games:

vigilante 8: still has massive graphical problems and sound issues
and
vigilante 8 second offence: still just crashes soon as i load the iso

**Comment by roflcopter777** [comment]
Tony Hawk's Pro Skater 1 and 2 hang with a black screen.

**Comment by MalphasX** [comment]
Are they by any chance ecm files? I've noticed my only two ecm games have those issues, and the other games all start and work just fine. Seems bin and img are the only ones supported.

**Comment by Orvis25** [comment]
@Mystro256 
Any new updated we can test? Hows it going?

**Comment by Orvis25** [comment]
@Mystro256 You alive D: ?

**Comment by Mystro256** [comment]
Lol yeah, just a busy time of year for me

**Comment by Orvis25** [comment]
@Mystro256 
Ah ok :) Well good luck and don't forget about us :D

**Comment by Restart80** [comment]
Not sure if anyone else is having this issue because it hasn't been mentioned I think. But my controller randomly becomes unresponsive for a few seconds on this emulator (if I'm moving in a direction, it will keep going for several seconds and I can't do anything, as if it is "locked"). Not sure if it's for all games, but it happens a lot on Digimon World 1.

**Comment by qamichele5959** [comment]
What is so happening from this game is not booting up on PSP with nulldce-psp emulator(?)
[https://github.com/PSP-Archive/nulldce-psp/issues/143](here)

**Comment by qamichele5959** [comment]
Will fail on latest wiisx build but works on older mednafenpsx core libretro wii build 1.0.0.3.

**Comment by qamichele5959** [comment]
> Tony Hawk's Pro Skater 1 and 2 hang with a black screen.
[Does have the DC version game will boot on PSP using the nulldce-psp emulator??](https://github.com/PSP-Archive/nulldce-psp/issues/149)

**Comment by qamichele594-2** [comment]
> Tony Hawk's Pro Skater 1 and 2 hang with a black screen.

When getting all work on PCSX Autobleem emulator after trying to buy a playstation classic console from year 2018, there has been crashing and major looping in both WiiSXRX and PCSX-Revolution emulators, but probably the PCSX-ReARMed libretro3ds core emulator has "prefetch abort (kernel panic)" and "prefetch abort (svcBreak)" and "The card was removed." and "data abort" and "undefined instruction" and "An error has occurred: Couldn't find any FIRM file(s). Press any button to shutdown" and "An error has occurred: Unable to mount CTRNAND or load the CTRNAND FIRM. Please use an external one. Press any button to shutdown" and "An error has occurred: Failed to mount CTRNAND. Press any button to shutdown" and "An error has occurred: Failed to apply 1 FIRM patch(es). Press any button to shutdown" and "An error has occurred: Failed to apply 2 FIRM patch(es). Press any button to shutdown" and "An error has occurred: Failed to apply 3 FIRM patch(es). Press any button to shutdown" and "An error has occurred: Failed to apply 5 FIRM patch(es). Press any button to shutdown" and "An error has occurred: Failed to apply 4 FIRM patch(es). Press any button to shutdown" and "An error has occurred: Failed to apply 6 FIRM patch(es). Press any button to shutdown" and "An error has occurred: Failed to get the CTRNAND FIRM. Press any button to shutdown" and "An error has occurred: Failed to find the CTRNAND FIRM. Press any button to shutdown" and "An error has occurred: Failed to decrypt the CTRNAND FIRM. Press any button to shutdown" and "An error has occurred: Failed to encrypt the CTRNAND FIRM. Press any button to shutdown" and "An error has occurred: Failed to reinsert the CTRNAND FIRM. Press any button to shutdown" and "An error has occurred: The CTRNAND FIRM is corrupted. Press any button to shutdown" and "An error has occurred: Unsupported launcher (argc = 0). Press any button to shutdown" errors and the game still NOT working anymore in newest PCSX-ReARMed libretro3DS core emulator version 1.10.0 on your O2DS console, and probably the ePSXe emulator has "ePSXe.exe has stopped working. A problem caused the program to stop working correctly. Windows will close the program and notify you if a solution is available." and "Error running ePSXe: SPEC Opcode 1e UNK (PC 00000134) (00000134) (264,45)" and "Error running ePSXe: unknown opcode 00103e44 at 08000345" and "Error running ePSXe: ePSXe (error) recompile block too large (28673)(28672)" and "Error running ePSXe: unknown opcode 00a11235 at 00cda4fe" errors and the game still NOT working anymore again in latest ePSXe release 2.0.5 on my PC Computer, just gives an error:
```
 
 An exception occurred
 
 Processor:       ARM9
 Exception type:  prefetch abort (kernel panic)
 
 R0     00000000       R1     0811FFFF
 R2     00000084       R3     00000000
 R4     080FE8D0       R5     00000FEE
 R6     00800DFE       R7     0000030E
 R8     00000000       R9     00000000
 R10    000036C0       R11    0000000B
 R12    00000008       SP     080EEA0A
 LR     08039455       PC     08075522
 CPSR   60000030
 
 You can find a dump in the following file:
 dumps/arm9/crash_dump_00000012.dmp
 
 Press any button to shutdown
```
And
```
 
 An error occurred (ErrDisp)
 
 Error type:       generic
 
 Process ID:       67115840
 
 Address:          0x00000e1e
 Error code:       0x000000e1
 
 
 Press any button to reboot.
```
And
```
 
 An error occurred (ErrDisp)
 
 Error type:       card removed
 
 The card was removed.
 
 
 Press any button to reboot.
```
And
```
 
 An error has occurred:
 
 Failed to encrypt the CTRNAND FIRM.
 
 Press any button to shutdown
```
And
```
 
 An error has occurred:
 
 Failed to mount CTRNAND.
 
 Press any button to shutdown
```
And
```
 
 An error has occurred:
 
 Unsupported launcher (argc = 0).
 
 Press any button to shutdown
```
And
```
 
 An exception occurred
 
 Processor:       ARM11 (core 0)
 Exception type:  prefetch abort (svcBreak)
 Fault status:    Debug event
 Current process: PCSX ReA (000400000BAC1500)
 
 R0     E0E01BF5       R1     00000000
 R2     0FFFFFE3       R3     00000000
 R4     00001000       R5     00100017
 R6     00023445       R7     000E0000
 R8     0000E000       R9     00AA0E00
 R10    0A00E0F0       R11    0000E000
 R12    00000000       SP     0AEEEFDD
 LR     00130122       PC     0015FE00
 CPSR   A0000010       FPEXC  00000000
 
 You can find a dump in the following file:
 dumps/arm11/crash_dump_00000018.dmp
 
 Press any button to shutdown
```
And
```
A problem has been detected and Windows has been shut down to prevent damage
to your computer.

The problem seems to be caused by the following file: ntoskrnl.exe

CRITICAL_PROCESS_DIED

If this is the first time you've seen this stop error screen,
restart your computer. If this screen appears again, follow
these steps:

Check to make sure any new hardware or software is properly installed.
If this is a new installation, ask your hardware or software manufacturer
for any Windows updates you might need.

If problems continue, disable or remove any newly installed hardware
or software. Disable BIOS memory options such as caching or shadowing.
If you need to use safe mode to remove or disable components, restart
your computer, press F8 to select Advanced Startup Options, and then
select Safe Mode.

Technical Information:

*** STOP: 0x000000f0 (0xfffff78000509518, 0x0000000000000000, 0xfffff90002e64beb, 
0x0000000000000003)

*** ntoskrnl.exe - Address 0xfffff80002e8a5e2 base at 0xfffff80002e07000 DateStamp 
0x55afca65
```
And
`ePSXe.exe has stopped working`
And
`Error running ePSXe: SPEC Opcode 1e UNK (PC 00003456) (00003456) (234,236)`
And
`Error running ePSXe: unknown opcode 00103455 at 039945fe`

**Comment by qamichele594-2** [comment]
[Somehow is dead?](https://github.com/niuus/WiiSXRX/issues/58)
